res_partner_fiscal_document/res_partner.py

```diff
@@ -0,0 +1,151 @@
+# -*- encoding: utf-8 -*-
+# #############################################################################
+#
+# OpenERP, Open Source Management Solution
+# Copyright (C) Odoo Colombia (Community).
+# Author        David Arnold (devCO)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api # , _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    dom = "[  '|',                                " \
+          "    ('on_company',  '=', is_company),  " \
+          "    ('on_contact', '!=', is_company)  ]"
+    fiscal_id_type = fields.Many2one(
+        'res.partner.idtype',
+        string=u'Document Type',
+        domain=dom,
+    )
+    fiscal_id = fields.Char(
+        string=u'Document ID',
+    )
+    fiscal_id_doc = fields.Binary(
+        string=u'Document Scan',
+        help="Upload the supporting Document "
+             "preferably as size-optimized PDF. "
+             "This might "
+             "help save disk space and PDF allows "
+             "you to concatenate multiple documents."
+    )
+
+    @api.one
+    @api.onchange(
+        'fiscal_id_type',
+        'fiscal_id',
+        'is_company',
+    )
+    def validateformatcopy(self):
+        # CASE: Current ID Type is not applicable on company
+        if self.is_company and not self.fiscal_id_type.on_company:
+            # Get the first valid ID type (remember: ordered by sequence)
+            self.fiscal_id_type = self.env['res.partner.idtype'].search(
+                [('on_company', '=', True)], limit=1).id
+            self.fiscal_id = None  # Reset ID value
+        # CASE: Current ID Type is not applicable on contact
+        if not self.is_company and not self.fiscal_id_type.on_contact:
```

Comment:
probably should be `elif` instead of `if` to make it clear that the conditions are mutually exclusive.